### PR TITLE
feature: header items should be links

### PR DESCRIPTION
**Pull request by buzzub**
Resolves [158](https://github.com/UKHO/admiralty-design-system/issues/158)

The links in the header / navigation menu component are rendered using buttons that emit events. For better accessibility, links should be rendered using `<a>` rather than a `<button>`.

To ensure compatibility with existing implementations, a new component `<AdmiraltyHeaderMenuLink/>` has been created. This accepts a `href` prop and will render an `<a>` tag. This will give the appearance of a link for users (hover shows the target) and correctly reads out a link for screen readers.

For compatibility with frameworks that use routers for navigation e.g. `router.push('/home')` the default browser redirect behaviour for the link can be suppressed using the `suppressRedirect` prop. When the link is clicked the `menuItemClick` event is emitted and the router navigation can be used instead of forcing the browser to do a full page reload.

<details>
  <summary>📦 Published PR as canary version: <code>0.21.0--canary.233.2642994.0</code></summary>
  <br />

  :sparkles: Test out this PR locally via:
  
  ```bash
  npm install @ukho/admiralty-angular@0.21.0--canary.233.2642994.0
  npm install @ukho/admiralty-core@0.21.0--canary.233.2642994.0
  # or 
  yarn add @ukho/admiralty-angular@0.21.0--canary.233.2642994.0
  yarn add @ukho/admiralty-core@0.21.0--canary.233.2642994.0
  ```
</details>

### CODE REVIEW

File: packages/website/src/usage/header/menu-item/javascript.mdx
Comment (UKHOGreg) [inline packages/website/src/usage/header/menu-item/javascript.mdx:1]: Just want to check this

File: packages/website/src/usage/header/menu-item/angular.mdx
Comment (UKHOGreg) [inline packages/website/src/usage/header/menu-item/angular.mdx:1]: Just want to check this

File: packages/website/src/usage/header/menu-item/javascript.mdx [hunk 1]
```diff
@@ -1,15 +1,29 @@
...
+    <admiralty-header-sub-menu-item
+      menu-title="Super long sub navigation item for wrapping"
+    ></admiralty-header-sub-menu-item>
     <admiralty-header-sub-menu-item menu-title="sub item 3"></admiralty-header-sub-menu-item>
-    </admiralty-header-menu-item>
-  <admiralty-header-menu-item menu-title="Components &amp; Patterns" active="false" slot="items">
+  </admiralty-header-menu-item>
+  <admiralty-header-menu-link
+    menu-title="Components & Patterns"
+    active="{false}"
```
Comment (UKHOGreg) [inline packages/website/src/usage/header/menu-item/javascript.mdx:18]:
```suggestion
    active="false"
```

File: packages/website/src/usage/header/menu-item/javascript.mdx [hunk 2]
```diff
@@ -1,15 +1,29 @@
...
+    href="/components/header"
+    suppress-redirect="{true}"
```
Comment (UKHOGreg) [inline packages/website/src/usage/header/menu-item/javascript.mdx:20]:
```suggestion
    suppress-redirect="true"
```

File: packages/website/src/usage/header/menu-item/angular.mdx [hunk 3]
```diff
@@ -1,15 +1,29 @@
...
+    <admiralty-header-sub-menu-item
+      menu-title="Super long sub navigation item for wrapping"
+    ></admiralty-header-sub-menu-item>
     <admiralty-header-sub-menu-item menu-title="sub item 3"></admiralty-header-sub-menu-item>
-    </admiralty-header-menu-item>
-  <admiralty-header-menu-item menu-title="Components &amp; Patterns" active="false" slot="items">
+  </admiralty-header-menu-item>
+  <admiralty-header-menu-link
+    menu-title="Item 2"
+    active="{false}"
```
Comment (UKHOGreg) [inline packages/website/src/usage/header/menu-item/angular.mdx:18]:
```suggestion
    active="false"
```

File: packages/website/src/usage/header/menu-item/angular.mdx [hunk 4]
```diff
@@ -1,15 +1,29 @@
...
+    href="/components/header"
+    suppress-redirect="{true}"
```
Comment (UKHOGreg) [inline packages/website/src/usage/header/menu-item/angular.mdx:20]:
```suggestion
    suppress-redirect="true"
```

**Review by UKHOGreg — CHANGES_REQUESTED** [review]

File: packages/website/src/usage/header/menu-item/javascript.mdx
Comment (UKHOGreg) [reply packages/website/src/usage/header/menu-item/javascript.mdx:1]: I have the same issue with the latest javascript example

**Review by UKHOGreg — CHANGES_REQUESTED** [review]

File: packages/website/src/usage/header/menu-item/javascript.mdx [hunk 5]
```diff
@@ -1,15 +1,29 @@
...
-    </admiralty-header-menu-item>
-  <admiralty-header-menu-item menu-title="Components &amp; Patterns" active="false" slot="items">
+  </admiralty-header-menu-item>
+  <admiralty-header-menu-link
+    menu-title="Item 2"
+    active="false"
+    href="/components/header"
+    suppress-redirect="true"
+    slot="items"
+  />
```
Comment (UKHOGreg) [inline packages/website/src/usage/header/menu-item/javascript.mdx:22]:
```suggestion
    slot="items">
  </admiralty-header-menu-link>
```

**Review by UKHOGreg — DISMISSED** [review]

**Review by ecstaticrainbow — APPROVED** [review]
LGTM

**Pull request merged**